src/main/groovy/org/shipkit/internal/gradle/versionupgrade/CreatePullRequestTask.java

```diff
@@ -0,0 +1,142 @@
+package org.shipkit.internal.gradle.versionupgrade;
+
+import org.gradle.api.DefaultTask;
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
+import org.gradle.api.tasks.TaskAction;
+import org.shipkit.gradle.ReleaseConfiguration;
+import org.shipkit.internal.util.ExposedForTesting;
+import org.shipkit.internal.util.GitHubApi;
+
+import java.io.IOException;
+
+/**
+ * Creates a pull request in {@link CreatePullRequestTask#repositoryUrl} of title {@link CreatePullRequestTask#title}
+ * between {@link CreatePullRequestTask#baseBranch} and {@link CreatePullRequestTask#headBranch}
+ */
+public class CreatePullRequestTask extends DefaultTask{
+
+    private static final Logger LOG = Logging.getLogger(CreatePullRequestTask.class);
+
+    private String repositoryUrl;
+    private String gitHubApiUrl;
+    private String authToken;
+    private String title;
+    private String headBranch;
+    private String baseBranch;
+
+    private GitHubApi gitHubApi;
+
+    @TaskAction
+    public void createPullRequest() throws IOException {
+        if(gitHubApi == null) {
+            gitHubApi = new GitHubApi(gitHubApiUrl, authToken);
+        }
+
+        LOG.lifecycle("  Creating a pull request of title '{}' in repository '{}' between base = '{}' and head = '{}'.",
+            title, repositoryUrl, baseBranch, headBranch);
+
+        String body = "{" +
+            "  \"title\": \"" + title + "\"," +
+            "  \"body\": \"Please pull this in!\"," +
+            "  \"head\": \"" + headBranch + "\"," +
+            "  \"base\": \"" + baseBranch + "\"" +
+            "}";
+
+        try {
+            gitHubApi.post("/repos/" + repositoryUrl + "/pulls", body);
+        } catch(IOException e){
+            LOG.error("  Creating a pull request failed.\n  {}", e.getMessage());
```

Comment:
Maybe you should throw RuntimeException or something which extened RuntimeException? The question is. If post throw exception the task should be interrupt?